cpp/src/arrow/testing/gtest_util.h

```diff
@@ -108,9 +108,9 @@
   } while (false);
 
 #define ASSIGN_OR_HANDLE_ERROR_IMPL(handle_error, status_name, lhs, rexpr) \
-  auto status_name = (rexpr);                                              \
+  auto&& status_name = (rexpr);                                            \
   handle_error(status_name.status());                                      \
-  lhs = std::move(status_name).ValueOrDie();
+  lhs = std::move(status_name).ValueUnsafe();
```

Comment:
Isn't this a problem with `EXPECT_OK_AND_ASSIGN`? Basically `lhs` will get some undetermined value.

Reply:
Good point. I'll change this one back to ValueOrDie